Library/Homebrew/extend/os/linux/parser.rb

```diff
@@ -15,7 +15,9 @@ def validate_options
         return unless @args.respond_to?(:cask?)
         return unless @args.cask?
 
-        raise UsageError, "Casks are not supported on Linux"
+        # NOTE: We don't raise a UsageError here because
+        # we don't want to print the help page.
+        raise "Invalid usage: Casks are not supported on Linux"
```

Comment:
```suggestion
        raise ArgumentError, "Invalid `--cask` usage: Casks do not work on Linux"
```
or
```suggestion
        odie "Invalid `--cask` usage: Casks do not work on Linux"
```
- want to avoid bare `RuntimeException` where possible here
- let's note that passing `--cask` was the problem
- let's use something stronger/different from "supported" as e.g. building from source is "not supported on Linux"